sqlglot/optimizer/qualify_columns.py

```diff
@@ -0,0 +1,321 @@
+        (Same as `_qualify_union`)
+    Returns:
+        (Same as `_qualify_union`)
+    """
+    # pylint: disable=too-many-locals
+    ctes = []
+    derived_tables = []
+    subqueries = []
+    tables = []
+    columns = []
+    selections = []
+    select_stars = []
+
+    # Collect all the selections in this context
+    for selection in expression.args.get("expressions", []):
+        if isinstance(selection, exp.Star) or (
+            isinstance(selection, exp.Column) and isinstance(selection.this, exp.Star)
+        ):
+            select_stars.append(selection)
+        else:
+            selections.append(selection)
+
+    # Collect all the other relevant nodes in this context
+    # Only traverse down to the next SELECT statements - we'll recurse into those later
+    for node, parent, _ in expression.walk(stop_types=exp.Select):
+        if node is expression:
+            continue  # Skip this node itself - we only care about children
+        if isinstance(node, exp.CTE):
+            ctes.append(node)
+        elif isinstance(node, (exp.Select, exp.Union)):
+            if isinstance(parent, exp.Subquery):
+                derived_tables.append(parent)
+            else:
+                subqueries.append(node)
+        elif isinstance(node, exp.Table):
+            tables.append(node)
+        elif isinstance(node, exp.Column):
+            columns.append(node)
+
+    parent_selectables = _merge(
+        parent_selectables,
+        _qualify_derived_tables(ctes, schema, sequence, parent_selectables, chain=True),
+    )
+    selectables = _merge({}, _qualify_tables(tables, schema, parent_selectables))
+    selectables = _merge(
+        selectables,
+        _qualify_derived_tables(derived_tables, schema, sequence, parent_selectables),
+    )
+    expansions = _expand_stars(select_stars, selectables)
+    columns.extend(expansions)
+    selections.extend(expansions)
+    selectables = _merge(parent_selectables, selectables)
+    _qualify_columns(columns, selectables)
+    _qualify_subqueries(subqueries, schema, sequence, selectables)
+    return _qualify_outputs(selections)
+
+
+def _qualify_derived_tables(
+    derived_tables, schema, sequence, parent_selectables, chain=False
+):
+    selectables = {}
+    for subquery in derived_tables:
+        subquery_outputs = _qualify_union(
+            expression=subquery.this,
+            schema=schema,
+            sequence=sequence,
+            parent_selectables=_merge(parent_selectables, selectables)
+            if chain
+            else parent_selectables,
+        )
+
+        table_alias = subquery.args.get("alias")
+        if not table_alias:
+            table_alias = exp.TableAlias()
+            subquery.set("alias", table_alias)
+
+        alias = table_alias.args.get("this")
+        if not alias:
+            alias = exp.to_identifier(f"_alias{sequence.next()}")
+            table_alias.set("this", alias)
+
+        alias_columns = table_alias.args.get("columns")
```

Comment:
ah this is alias_columns, yea sorry, confused. i don't think these add any values as a concept